Diffusion/Common/ImportModelBehavior.swift

```diff
@@ -0,0 +1,97 @@
+                            isBadSelectionAlertShown = true
+                        }
+                    } else if url.pathExtension == "zip" {
+                        DispatchQueue.global(qos: .background).async {
+                            do {
+                                let extractedURL = Settings.shared.applicationSupportURL().appendingPathComponent("hf-diffusion-models").appendingPathComponent(url.deletingPathExtension().lastPathComponent)
+                                try FileManager.default.createDirectory(at: extractedURL, withIntermediateDirectories: true, attributes: nil)
+
+                                if let archive = Archive(url: url, accessMode: .read) {
+                                    for entry in archive {
+                                        // Skip hidden and __MACOSX files
+                                        let filename = entry.path.components(separatedBy: "/").last ?? ""
+                                        if !filename.hasPrefix(".") && filename.lowercased() != "__macosx" {
+                                            let destinationURL = extractedURL.appendingPathComponent(entry.path)
+                                            let _ = try archive.extract(entry, to: destinationURL)
+                                        }
+                                    }
+                                }
+
+                                let extractedContents = try FileManager.default.contentsOfDirectory(atPath: extractedURL.path)
+                                if extractedContents.contains(where: { $0.caseInsensitiveCompare("merges.txt") == .orderedSame }) && extractedContents.contains(where: { $0.caseInsensitiveCompare("vocab.json") == .orderedSame }) {
+                                    // Files successfully extracted
+                                } else {
+                                    // Delete the extraction and post an alert that it's not properly formatted
+                                    try? FileManager.default.removeItem(at: extractedURL)
+                                    isBadSelectionAlertShown = true
+                                }
+                            } catch {
+                                print("Error unzipping selected zip file: \(error)")
+                            }
+                        }
+                    } else {
+                        // Invalid selection
+                        isBadSelectionAlertShown = true
+                    }
+                    
+                } catch {
+                    // Error handling
+                    print("Error: \(error.localizedDescription)")
+                }
+            }
+            .alert(isPresented: $isBadSelectionAlertShown) {
+                Alert(
+                    title: Text("Not a model"),
+                    message: Text("The selected folder does not appear to be a model. Please select an extracted model folder or a zip compressed model file."),
```

Comment:
```suggestion
                    title: Text("Invalid model"),
                    message: Text("The selected folder does not appear to contain a Core ML Stable Diffusion model. Please, select an extracted model folder or a zip compressed model file."),
```

People will try to import all sorts of things, including PyTorch models :)